src/widget/wsearchlineedit.cpp

```diff
@@ -309,6 +359,38 @@ void WSearchLineEdit::slotTriggerSearch() {
     emit search(getSearchText());
 }
 
+/// saves the current query as selection
+void WSearchLineEdit::slotSaveSearch() {
+    int cIndex = findData(currentText(), Qt::DisplayRole);
+    qDebug() << "save search. Index: " << cIndex;
+    m_saveTimer.stop();
+    // entry already exists and is on top
+    if (cIndex == 0) {
+        return;
+    }
+    if (currentText().length() && isEnabled()) {
```

Comment:
Use !isEmpty() instead of length() to be more explicit. Moreover, why an early return above but not here?